spec/models/payment_spec.rb

```diff
@@ -32,4 +33,61 @@
       expect(payment.amount_to_pay).to eq(8000)
     end
   end
+
+  describe '#purchase' do
+    let!(:user) { create(:user) }
+    let!(:conference) { create(:conference) }
+    let!(:ticket_1) { create(:ticket, price: 10, price_currency: 'USD', conference: conference) }
+    let!(:tickets) { {ticket_1.id.to_s => '2'} }
+    let(:stripe_helper) { StripeMock.create_test_helper }
+
+    before { StripeMock.start }
+    after { StripeMock.stop }
+
+    before { TicketPurchase.purchase(conference, user, tickets) }
+    let!(:payment) { create(:payment, user: user, conference: conference, stripe_customer_token: stripe_helper.generate_card_token, stripe_customer_email: user.email) }
+
+    context 'when the payment is successful' do
+      before { payment.purchase }
+
+      it 'assigns amount' do
+        expect(payment.amount).to eq(2000)
+      end
+
+      it 'assigns last4' do
+        expect(payment.last4).to eq('4242')
+      end
+
+      it "assigns 'success' to payment.status" do
+        expect(payment.status).to eq('success')
+      end
+
+      it 'assigns authorization_code' do
+        expect(payment.authorization_code).to eq('test_ch_3')
+      end
+    end
+
+    context 'if the payment is not successful' do
+      before { StripeMock.prepare_card_error(:invalid_number) }
+
+      let(:payment) { create(:payment, user: user, conference: conference, stripe_customer_token: 'bogus_card_token', stripe_customer_email: user.email) }
+
+      before { payment.purchase }
+
+      context 'when the card is invalid' do
```

Comment:
Can the gem also simulate a TimeOut? Are there any other errors we should consider?